src/Components/ChangePassword/ChangePassword.tsx

```diff
@@ -1,73 +1,135 @@
+import { useRef } from "react";
+import { useForm } from "react-hook-form";
+import AlertMessage from "../../Shared/AlertMessage/AlertMessage";
 
 export default function ChangePassword() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm();
+  const password = useRef({});
```

Comment:
This seems to me a lot of work, making the code harder to read and understand.
You can instead use `validate` and `getValues` from useForm hook and you access the any value in the form and validate if the newPassword matches the the confirmPassword.
here is an example that shows that:
```js
<input
              {...register("confirmNewPassword", {
                required: "This field is required",
                validate: (value) =>
                  value === getValues("newPassword") ||
                  "The passwords do not match",
              })}
                    placeholder="Confirm New Password"
/>
```